src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php

```diff
@@ -79,16 +95,111 @@ public function __construct(ProjectConfigInterface $projectConfig)
     public function getConfig()
     {
         $experimentsMaps = $this->getExperimentsMaps();
-        $featuresMap = $this->getFeaturesMap($experimentsMaps[1]);
-
+        $featuresMap = $this->getFeaturesMap($experimentsMaps[1], $this->project_config);
+        $attributes = $this->getConfigAttributes();
+        $audiences = $this->getConfigAudiences();
+        $events = $this->getConfigEvents();
         return new OptimizelyConfig(
             $this->revision,
             $experimentsMaps[0],
             $featuresMap,
-            $this->datafile
+            $this->datafile,
+            $this->environment_key,
+            $this->sdk_key,
+            $attributes,
+            $audiences,
+            $events
         );
     }
     
+    /**
+     * Generates array of attributes as OptimizelyAttribute.
+     *
+     *
+     * @return array of OptimizelyAttributes.
+     */
+    protected function getConfigAttributes()
+    {
+        $attribut_arr = [];
+        $attributes = $this->project_config->getAttributes();
+        foreach($attributes as $attr){
+            $id = $attr["id"];
+            $key = $attr["key"];
+            $optly_attr = new OptimizelyAttribute(
+                $id,
```

Comment:
directly use $attr["id"]; here instead of assigning both id and key to local variables first